packages/flutter/test/rendering/proxy_box_test.dart

```diff
@@ -454,6 +454,18 @@ void main() {
     // transform -> clip
     _testFittedBoxWithClipRectLayer();
   });
+
+  test('RenderFractionalTranslation updates its semantics after its translation value is set', () {
```

Comment:
Might be worthwhile to also add an actual semantics test to the FractionalTranslation widget that verifies that the generated semantics actually changed as expected when the translation is changed.

Reply:
I am having trouble writing such a test. I have:

```
    testWidgets('semantics bounds are updated', (WidgetTester tester) async {
      final GlobalKey fractionalTranslationKey = GlobalKey();
      final GlobalKey sizedBoxKey = GlobalKey();
      Offset offset = const Offset(0.4, 0.4);

      await tester.pumpWidget(
        StatefulBuilder(
          builder: (BuildContext context, StateSetter setState) {
            return Center(
              child: FractionalTranslation(
                key: fractionalTranslationKey,
                translation: offset,
                transformHitTests: true,
                child: Listener(
                  onPointerDown: (PointerDownEvent event) {
                    setState(() {
                      offset = const Offset(0.8, 0.8);
                    });
                  },
                  child: SizedBox(
                    key: sizedBoxKey,
                    width: 100.0,
                    height: 100.0,
                    child: Container(
                      color: const Color(0xFF0000FF)
                    ),
                  ),
                ),
              ),
            );
          },
        )
      );
      expect(tester.getSemantics(find.byKey(fractionalTranslationKey)), matchesSemantics(
        size: const Size(40, 40),
      ));
```

which fails with:

```
The following TestFailure object was thrown running a test:
  Expected: has semantics with actions: [] with flags: [] with rect: Rect.fromLTRB(0.0, 0.0, 0.0,
0.0)
  Actual: SemanticsNode:<SemanticsNode#0(Rect.fromLTRB(0.0, 0.0, 2400.0, 1800.0))>
   Which: rect was: Rect.fromLTRB(0.0, 0.0, 2400.0, 1800.0)
```

No matter what the translation is set to, the bounds are always (0.0, 0.0, 2400.0, 1800.0) for both the FractionalTranslation and the SizedBox inside

Reply:
After the pump can you call `debugDumpSemanticsTree` and post the output? That would make it easier to debug this.

My guess is that the the semantics of the FractionalTranslation get pulled up all the way into the root node and that's why you're seeing this. If you throw in a `Semantics(explicitChildNodes: true, child: ...)` around the FractionalTranslation or the Center widget it may work. Looking at the semantics tree dump would tell us for sure.

Reply:
output of `debugDumpSemanticsTree` is the same with or without the `Semantics(explicitChildNodes: true)` wrapper:

```
SemanticsNode#0
   Rect.fromLTRB(0.0, 0.0, 2400.0, 1800.0)
```

Reply:
That's the only node that gets created? 

Oh, it's probably because no child of FractionalTranslation has any real semantics to contribute. Try giving the container a `Text('foo')` child.

Reply:
```
SemanticsNode#0
 │ Rect.fromLTRB(0.0, 0.0, 2400.0, 1800.0)
 │
 └─SemanticsNode#1
     Rect.fromLTRB(0.0, 0.0, 100.0, 100.0) with transform
     [3.0,0.0,0.0,1170.0; 0.0,3.0,0.0,870.0; 0.0,0.0,1.0,0.0;
     0.0,0.0,0.0,1.0]
     label: "foo"
     textDirection: ltr
```